src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -1519,4 +1519,22 @@
     <value>and all other platforms</value>
     <comment>This call site is reachable on: 'windows' 10.0.2000 and later, and all other platforms</comment>
   </data>
-</root>
\ No newline at end of file
+  <data name="DoNotUseWaitAllWithSingleTaskDescription" xml:space="preserve">
+    <value>Using 'WaitAll' with a single task should be avoided in favor of directly awaiting that task or returning it as is.</value>
```

Comment:
Can we also include the `why` part in the description?

Reply:
Is it enough to say something like 

"Using 'WaitAll' may result in performance loss, await or return the task instead" ?

Reply:
That sounds fine to me.